src/newbank/server/NewBank.java

```diff
@@ -1,49 +1,128 @@
 
 public class NewBank {
-		
-	private customerDB customers;
-	
-	public NewBank() {
-		customers = new customerDB();
-	}
-	
-	public synchronized Customer checkLogInDetails(String customerID, String password) {
-		Customer customer = customers.getCustomer(customerID);
-		if (customer == null) {
-			return null;
-		} else if ( customer.checkPassword(password) ) {
-			return customer;
-		} else {
-			return null;
-		}
-	}
-
-	// commands from the NewBank customer are processed in this method
-	public synchronized String processRequest(String customerID, String request) {
-		Customer customer = customers.getCustomer(customerID);
-		if( customer != null ) {
-			if (request.equals("SHOWMYACCOUNTS")) {
-				return showMyAccounts(customer );
-			} else if (request.startsWith("NEWACCOUNT")) {
-				String accountName = request.substring( request.indexOf(" ")+1);
-				return makeNewAccount(customer, accountName);
-			}
-		}
-		return "FAIL";
-	}
-	
-	private String showMyAccounts(Customer customer) {
-		return customer.accountsToString();
-	}
-	
-	private String makeNewAccount(Customer customer, String accountName) {
-		customer.addAccount( new Account(accountName, 0.0) );
-		customers.updateCustomer(customer);
-		return "SUCCESS";
-	}
-	
 
+  private static final String SPACE = " ";
+
+  private CustomerDB customers;
+
+  public NewBank() {
+    customers = new CustomerDB();
+  }
+
+  public synchronized Customer checkLogInDetails(String customerID, String password) {
+    Customer customer = customers.getCustomer(customerID);
+    if (customer == null) {
+      return null;
+    } else if (customer.checkPassword(password)) {
+      return customer;
+    } else {
+      return null;
+    }
+  }
+
+  // commands from the NewBank customer are processed in this method
+  public synchronized String processRequest(String customerID, String request) {
+    Customer customer = customers.getCustomer(customerID);
+    if (customer != null) {
+      if (request.equals("SHOWMYACCOUNTS")) {
+        return showMyAccounts(customer);
+      } else if (request.startsWith("NEWACCOUNT")) {
+        String accountName = request.substring(request.indexOf(SPACE) + 1);
+        return makeNewAccount(customer, accountName);
+      } else if (request.startsWith("MOVE")) {
+        return moveMoney(customer, request.split(SPACE));
+      }
+    }
+    return "FAIL";
+  }
+
+  private String showMyAccounts(Customer customer) {
+    return customer.accountsToString();
+  }
+
+  private String makeNewAccount(Customer customer, String accountName) {
+    customer.addAccount(new Account(accountName, 0.0));
+    customers.updateCustomer(customer);
+    return "SUCCESS";
+  }
+
+  /**
+   * The command MOVE moves the money between accounts.
+   *
+   * @param customer to which moves the moeny.
+   * @param command MOVE.
+   * @return SUCCES or FAIL
+   */
+  // TODO method is to big should be refactored
+  private String moveMoney(Customer customer, String[] command) {
```

Comment:
Good to see details error messages. I will also try to implement details error messages like this for next stories. 

However as you might have already noticed, this is printing the error message in the server and not client. 
It'd have been really good if the end user/client got this message instead of server.

Reply:
I agree, but from the protocol, it was expected to provide either a FAIL or SUCCESS. We can discuss if we want to extend the protocol.